src/utils.ts

```diff
@@ -88,6 +88,17 @@ export function appendTsSuffixIfMatch(patterns: RegExp[], path: string): string
     return path;
 }
 
+export function appendTsxSuffixIfMatch(patterns: RegExp[], path: string): string {
```

Comment:
Having pondered I think it might make for nicer code to combine `appendTsSuffixIfMatch` and `appendTsxSuffixIfMatch`.  

Here's my reasoning:

Wherever `appendTsSuffixIfMatch` is called in the code, the output is then passed into the call to `appendTsxSuffixIfMatch`.  They are never called in isolation and are intended to be used together.  

That being the case, I think it might make sense to have an `appendTsTsxSuffixIfMatch` method which internally does the `appendTsSuffixIfMatch` and `appendTsxSuffixIfMatch` operations, feeding the output of the first into the second.  This should result in clearer code at the call sites.  It may also allow for some genericisation for the underlying methods (whether or not that makes sense I don't know)

Is that okay with you?  I'm quite focused on keeping the source of ts-loader as comprehensible as possible to ease contributions by others.